src/internal/connector/sharepoint/data_collections.go

```diff
@@ -0,0 +1,158 @@
+package sharepoint
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/onedrive"
+	"github.com/alcionai/corso/src/internal/connector/support"
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/observe"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/selectors"
+)
+
+type statusUpdater interface {
+	UpdateStatus(status *support.ConnectorOperationStatus)
+}
+
+type connector interface {
+	statusUpdater
+
+	Service() graph.Service
+	IncrementAwaitingMessages()
+}
+
+// DataCollections returns a set of DataCollection which represents the SharePoint data
+// for the specified user
+func DataCollections(
+	ctx context.Context,
+	selector selectors.Selector,
+	siteIDs []string,
+	tenantID string,
+	con connector,
+) ([]data.Collection, error) {
+	b, err := selector.ToSharePointBackup()
+	if err != nil {
+		return nil, errors.Wrap(err, "sharePointDataCollection: parsing selector")
+	}
+
+	var (
+		scopes      = b.DiscreteScopes(siteIDs)
+		collections = []data.Collection{}
+		serv        = con.Service()
+		errs        error
+	)
+
+	// for each scope that includes oneDrive items, get all
+	for _, scope := range scopes {
+		// Creates a slice of collections based on scope
+		dcs, err := createSharePointCollections(ctx, serv, scope, tenantID, con)
+		if err != nil {
+			return nil, support.WrapAndAppend(scope.Get(selectors.SharePointSite)[0], err, errs)
+		}
+
+		for _, collection := range dcs {
+			collections = append(collections, collection)
+		}
+	}
+
+	for range collections {
+		con.IncrementAwaitingMessages()
+	}
+
+	return collections, errs
+}
+
+// createSharePointCollections - utility function that retrieves M365
+// IDs through Microsoft Graph API. The selectors.SharePointScope
+// determines the type of collections that are retrieved.
+func createSharePointCollections(
```

Comment:
@ryanfkeepers - thoughts on implementing this similar to what `OneDrive` does? i.e. a `NewCollections()` and `Get` method? The logic in `DataCollections` above which has more GC specifics would move into `gc.SharePointCollections()`

I don't like for example exposing the internals of `con.IncrementAwaitingMessages()` here - that's very GC specific.

Reply:
Just to clarify - my suggestion for now is to move what is in `DataCollections` in this file to `func (gc *GraphConnector) SharePointDataCollections` and export `createSharePointCollections` as perhaps `sharepoint.NewCollections().Get(...)` (I'm hoping this is a pattern we can replicate for all the services)

RE: the duplication in `collectLibraries` - not worried about that for now - we can refactor later if needed.

Reply:
> Just to clarify...

Ahah, this is funny, because I just did _the very opposite_ in the [commit before you looked at the PR](https://github.com/alcionai/corso/pull/1536/commits/9f5f6dbceca0a94c7410955f3d44d323883ef19c).  My read on the `connector/data_collections.go` file is that we're packing in more service-specific concerns, and exercising a minimum of common gc-specific concerns (`incrementAwait` is one of the few).  And as we add cross-service interfaces and such, those service footprints grow larger while gc concerns remain static.

In all, it feels like this ownership is better housed within each service package, colocated with the other interfaces and functions used by that service.  But, this is not a strongly held opinion.  If you prefer to keep things in the collections package, I can stick with that.

Reply:
> My read on the connector/data_collections.go file is that we're packing in more service-specific concerns, and exercising a minimum of common gc-specific concerns (incrementAwait is one of the few)

Ah - I'm with you on that one. Service specific stuff under `sharepoint` and the move GC logic out into `func (gc *GraphConnector) SharePointDataCollections`